ecommerce_extensions/payment/pp_sandbox/sandbox_listener/views.py

```diff
@@ -0,0 +1,186 @@
+@permission_classes((permissions.AllowAny,))
+def transaction_pending(request):
+    if request.method == 'POST':
+        epp_request = request.POST.get('data')
+        epp_request = epp_request.replace("'", '"')
+        data = json.loads(epp_request)
+
+        #--------------------GET RESPONSE------------------
+        epp_get = get(data, "PENDING")
+        get_req = requests.get(data['responseUrl'],params = epp_get)
+
+        #-------------------POST RESPONSE------------------
+        epp_post = post(data, "PENDING")
+        post_req = requests.post(data['confirmationUrl'], data = epp_post)
+    
+    return redirect('http://localhost:18130/checkout/receipt/?order_number='+data['referenceCode'])
+
+@api_view(('GET', 'POST'))
+@permission_classes((permissions.AllowAny,))
+def transaction_declined(request):
+    if request.method == 'POST':
+        epp_request = request.POST.get('data')
+        epp_request = epp_request.replace("'", '"')
+        data = json.loads(epp_request)
+
+        #--------------------GET RESPONSE------------------
+        # epp_get = get(data, "DECLINED")
+        # get_req = requests.get(data['responseUrl'], params = epp_get)
+
+        #-------------------POST RESPONSE------------------
+        epp_post = post(data, "DECLINED")
+        post_req = requests.post(data['confirmationUrl'], data = epp_post)
+    
+    return redirect(data['confirmationUrl'])
+
+def post(data, response_type):
+
+    if response_type == 'APPROVED':
+        STATE_POL = "4"
+        RESPONSE_CODE_POL = "1"
+    elif response_type == 'PENDING':
+        STATE_POL = "7"
+        RESPONSE_CODE_POL = "25"
+    elif response_type == 'DECLINED':
+        STATE_POL = "6"
+        RESPONSE_CODE_POL = "104"
+
+    value = data['amount']
+    last_decimal = value[-1]
+    if last_decimal == '0':
+        new_value = value[:-1]
+    else:
+        new_value = value
+
+    uncoded = "{api_key}~{merchant_id}~{reference_sale}~{new_value}~{currency}~{state_pol}".format(
+        api_key=data['ApiKey'],
+        merchant_id=data['merchantId'],
+        reference_sale=data['referenceCode'],
+        new_value=new_value,
+        currency=data['currency'],
+        state_pol='4',
+    )
+    
+    sign = sha256(uncoded.encode('utf-8')).hexdigest()
+
+    id = uuid.uuid1()
+
+    post_res = {
+        "state_pol": STATE_POL,
+        "response_code_pol": RESPONSE_CODE_POL,
+        "reference_sale": data['referenceCode'],
+        "reference_pol": "7069375",
+        "sign": sign,
+        "value": data['amount'],
+        "currency": data['currency'],
+        "email_buyer": data['buyerEmail'],
+        "test": data['test'],
+        "description": "Test PP",
+        "billing_address": "Calle 123",
+        "billing_city": "Los Mochis",
+        "billing_country": "MX",
+        "transaction_id": id,
+        "response_message_pol": response_type,
+    }
+    return post_res
+
+def get(data, response_type):
+
+    if response_type == 'APPROVED':
```

Comment:
Since this exact same piece of code is in the post method, maybe you could create a function that returns the state_pol and response_code_pol values based on the response_type ?